Code/Source/Sensor/SensorConfiguration.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+* Copyright (c) Contributors to the Open 3D Engine Project.
+* For complete copyright and license terms please see the LICENSE at the root of this distribution.
+*
+* SPDX-License-Identifier: Apache-2.0 OR MIT
+*
+*/
+
+#pragma once
+
+#include "Sensor/SensorConfiguration.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+
+namespace ROS2
+{
+    void SensorConfiguration::Reflect(AZ::ReflectContext* context)
+    {
+        PublisherConfiguration::Reflect(context);
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->RegisterGenericType<AZStd::shared_ptr<PublisherConfiguration>>();
+            serializeContext->RegisterGenericType<AZStd::map<AZStd::string, AZStd::shared_ptr<PublisherConfiguration>>>();
+            serializeContext->Class<SensorConfiguration>()
+                ->Version(2)
+                ->Field("Visualise", &SensorConfiguration::m_visualise)
+                ->Field("Publishing Enabled", &SensorConfiguration::m_publishingEnabled)
+                ->Field("Frequency (HZ)", &SensorConfiguration::m_frequency)
+                ->Field("Publishers", &SensorConfiguration::m_publishersConfigurations)
+                ;
+
+            if (AZ::EditContext* ec = serializeContext->GetEditContext())
+            {
+                ec->Class<SensorConfiguration>("ROS2 Sensor Configuration", "Sensor configuration")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &SensorConfiguration::m_visualise, "Visualise", "Visualise")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &SensorConfiguration::m_publishingEnabled, "Publishing Enabled", "Toggle publishing for topic")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &SensorConfiguration::m_frequency, "Frequency", "Frequency of publishing")
```

Comment:
You mentioned `Hz` before, therefore I think it would be generally a good idea to include units in all descriptions (if it applies).

```suggestion
                    ->DataElement(AZ::Edit::UIHandlers::Default, &SensorConfiguration::m_frequency, "Frequency", "Frequency of publishing [Hz]")
```